2025 Survey Draft Updates

isabela-pf opened the pull request:
I'm proposing updates to the 2025 survey draft based on feedback in [sphinx-doc discussions #13331](https://github.com/orgs/sphinx-doc/discussions/13331). 

I've integrated a majority of the changes (though I imagine there may be feedback on the best way to do that) except for 

> I'd like to have insights into preference (How important is ...) and satisfaction (How satisfied are you with ...) on topics like
> 
> - Performance
> - Stability
> - Ease of use (Maybe subtopics: setup, learn, write, build, customize)
> - Sphinx own documentation
> - Default theme
> - Builtin themes
> - ReST vs. Markdown
> - Extension ecosystem
> - API documentation generation

Drafts of these questions or how they may be integrated into existing questions are not in this PR. This is something I'm still considering the best approach for, but I didn't want to keep holding up the other changes longer.

As always, feedback is very much welcome and I will do my best to answer any questions. Thank you for making this repository and being open to the survey!

Commit:
Update survey draft based on question-by-question feedback

trallard commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -32,19 +32,6 @@ questions:
-      - "Technical support"
-      - "Technical writer"
-      - "Prefer not to answer"
-
```
IIRC, from https://github.com/Quansight-Labs/typeform-yaml-survey/pull/4, this was to be kept.

For now, let's add this again, and we can sanitise these questions later on.

trallard reviewed (approved):
This looks good to me, so I will approve this, but we'll need @AA-Turner or someone else with commit rights to merge.

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -103,6 +90,11 @@ questions:
+      - "Astro"
+      - "Pelican"
+      - "Nikola"
+      - "Eleventy"
```
To be picky, can we sort these lists alphabetically?

@Carreau is it possible to enable choice order randomisation in the typeform survey?

Carreau replied:
It would need to be implemented.

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -113,7 +105,14 @@ questions:
+      - "Render"
+      - "Vultr"
       - "My institution has its own deployment"
+      - "I don't know"
```
What's the goal of the deployment question? I might suggest adding "as part of an existing website" (i.e. just uploading to a `/docs` directory).

Could we also tweak "my institution has its" to "we have our"? For personal projects, there's no institution, and e.g. if I was answering this with my Python committer hat on, I'm not sure I'd call docs.python.org and/or the infra team "my institution".

isabela-pf replied:
My understanding is that deployment question is part of building a sense of how all the infrastructural pieces of documentation writing fit together. Knowing more about how these pieces are put together might be useful in noticing potential pain points in the places where Sphinx gets hooked into other software. 

I do think it's more tangential to Sphinx than some of the others, so I could see an argument to remove it.

I'm glad you asked, because it seems important to ensure what sort of field of view seems fitting for this survey.

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -113,7 +105,14 @@ questions:
       - "My institution has its own deployment"
+      - "I don't know"
 
   - title: "What markup language are you using in your documentation?"
```
Could we add ASCIIdoc(ter), XML/DITA (if this is a language!), TeX, Confluence/Jira, and perhaps split out flavours of markdown (e.g. MyST vs GFM vs CommonMark)?

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -178,16 +177,17 @@ questions:
-      - "Python API documentation generation (docstrings)"
-      - "Narrative documentation"
-      - "Educational materials"
+      - "API documentation generation from docstrings"
```
I'd like to make sure we're not too Python specific (Sphinx is used by large non-Python projects such as Linux), so we should disambiguate when using Python jargon.
```suggestion
      - "API documentation generation from Python docstrings"
```

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -178,16 +177,17 @@ questions:
-      - "Narrative documentation"
-      - "Educational materials"
+      - "API documentation generation from docstrings"
+      - "Being able to reference to other pages/sections across one or multiple sites"
```
```suggestion
      - "Being able to cross-reference other pages/sections across one or multiple sites ('intersphinx')"
```

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -205,6 +205,8 @@ questions:
       - "Pydata Sphinx Theme"
       - "Material design"
+      - "Furo"
+      - "Python Docs Sphinx Theme"
```
* https://github.com/lepture/shibuya
* https://github.com/executablebooks/sphinx-book-theme
* https://github.com/jbms/sphinx-immaterial
* https://github.com/kai687/sphinxawesome-theme
* https://github.com/ryan-roemer/sphinx-bootstrap-theme
```suggestion
      - "Python Docs Sphinx Theme"
      - "Sphinx Book Theme"
      - "Shibuya"
      - "Sphinx Immaterial"
      - "Sphinx Awesome Theme"
      - "Sphinx Bootstrap Theme"
```

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -215,20 +217,6 @@ questions:
       - "PDF"
       - "EPUB"
```
Can we add manpages / manual pages? Perhaps also JSON serialisation, XML, Texinfo, and plain text?

AA-Turner commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -215,20 +217,6 @@ questions:
       - "PDF"
       - "EPUB"
 
-  - title: "How comfortable do you feel when working with docutils?"
-    choices:
-      - "Very comfortable"
-      - "Comfortable"
-      - "Neutral"
-      - "Uncomfortable"
-      - "Very uncomfortable"
-      - "I don't know or don't work with docutils"
```
For personal use, some of this information would be interesting, though we probably have better contact with extension developers than 'ordinary' users. It might be worth thinking about a conditional section behind "do you develop a sphinx extension"? This is somewhat marginal though, so if a less complex survey is wanted then I'm content with this.

isabela-pf replied:
I know there was discussion about why this was here and if it fit with the larger goals, so I thought it might be easiest to remove right now. I do know @Carreau has mentioned having more certainty that things will work right with less complex survey logic, so that is my goal right now too.

Carreau replied:
Another of my fear is better is the enemy of good; 

I thing one of our goal should be to run survey regularly, learn from results and iterate, and the more we try to be perfect the more we burnout people wanting to:

1) work on the survey
2) take the survey
3) analyze the survey

I also think that if there are a subset of users (say extension developers), we can at the end have a "are you an extension developers, we want to know more, here is another survey"

That way we still have not taken them too much time and "Trap" them in a longer survey.

isabela-pf committed:
Apply suggestions from @AA-Turner review

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>

Commit:
Alphabetize options and add options

Commit:
Add @trallard feedback

isabela-pf commented:
Thank you both for the review! I think I've addressed all the comments so far, notably re-adding a question, adding options to a number of questions, and alphabetizing them. I think this is ready for review again (or another next step).

Commit:
Add introductory comments and add Sphinx relation question option

trallard commented on surveys/sphinx-survey-2025.yml:
```diff
@@ -110,27 +122,41 @@ questions:
+      - "AsciiDoc"
+      - "Confluence/Jira"
+      - "Markdown - CommonMark"
+      - "Markdown - GitHub Flavored Markdown"
```
We are repeating markdown here. 
For consistency this should be changed to Markdown - GitHub
